onnxruntime/core/framework/allocation_planner.cc

```diff
@@ -643,7 +649,8 @@ class PlannerImpl {
         if (node_input->Exists()) {
           auto& sym = node_input->Name();
           auto original = Buffer(Index(sym));
-          if (0 == --UseCount(original))
+          // The index will be -1 if it's an initializer that was removed as part of an optimization
+          if ((original != -1) && (0 == DecrementUseCount(original)))
```

Comment:
Given it's -1 due to a hack involving a const_cast (i.e. the ORT API is being explicitly circumvented and the ORT code shouldn't be expected to handle the result of that), it may be clearer to say that this a temporary workaround for an issue that we need to fix asap. The current comment would make me look at the optimizers library to try and understand how we could possibly 'remove' an initializer but still have a node that we're going to execute relying on it.